### PR TITLE
[chore]ddtrace/tracer: added a WithAgentTimeout option to reduce the test suite duration with no agent

### PR DESCRIPTION
**Pull request by dianashevchenko**
### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: ddtrace/internal/globaltracer.go [hunk 1]
```diff
@@ -39,6 +39,10 @@ func GetGlobalTracer() ddtrace.Tracer {
 // environment. This value is used by tracer.Start to prevent overriding the GlobalTracer in tests.
 var Testing = false
 
+// TestingWithAgent is set to true when we expect to load Agent features. This is used to imrpove dev experience
```
Comment (darccio) [inline ddtrace/internal/globaltracer.go:42]:
```suggestion
// TestingWithAgent is set to true when we expect to load Agent features. This is used to improve dev experience
```

File: ddtrace/tracer/meta_struct_test.go [hunk 2]
```diff
@@ -8,10 +8,11 @@ package tracer
 import (
 	"bytes"
 	"errors"
-	"github.com/stretchr/testify/require"
 	"reflect"
 	"testing"
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/tinylib/msgp/msgp"
```
Comment (darccio) [inline ddtrace/tracer/meta_struct_test.go:16]:
```suggestion
	"github.com/stretchr/testify/require"
	"github.com/tinylib/msgp/msgp"
```

File: ddtrace/tracer/metrics_test.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ import (
 	globalinternal "gopkg.in/DataDog/dd-trace-go.v1/internal"
 
 	"github.com/stretchr/testify/assert"
+
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/statsdtest"
```
Comment (darccio) [inline ddtrace/tracer/metrics_test.go:16]:
```suggestion
	globalinternal "gopkg.in/DataDog/dd-trace-go.v1/internal"
	"gopkg.in/DataDog/dd-trace-go.v1/internal/statsdtest"

	"github.com/stretchr/testify/assert"
```

File: ddtrace/tracer/option.go [hunk 4]
```diff
@@ -22,6 +22,8 @@ import (
 	"strings"
 	"time"
 
+	traceinternal "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
+
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
```
Comment (darccio) [inline ddtrace/tracer/option.go:28]:
```suggestion
	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
	traceinternal "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
```

File: ddtrace/tracer/option_test.go [hunk 5]
```diff
@@ -98,6 +100,7 @@ func TestStatsdUDPConnect(t *testing.T) {
 }
 
 func TestAutoDetectStatsd(t *testing.T) {
+	internal.TestingWithAgent = true
```
Comment (darccio) [inline ddtrace/tracer/option_test.go:103]: Shouldn't this kind of calls have a `defer` statement to revert the global variable to its original value? Also, I think it should be done only in the places where we expect `TestingWithAgent` to be `false` if the default is `true` (although I see that you initialized it to `false` in the `TestMain`).

File: internal/datastreams/transport.go [hunk 6]
```diff
@@ -10,46 +10,16 @@ import (
...
-		Proxy:                 http.ProxyFromEnvironment,
-		DialContext:           defaultDialer.DialContext,
-		MaxIdleConns:          100,
-		IdleConnTimeout:       90 * time.Second,
-		TLSHandshakeTimeout:   10 * time.Second,
-		ExpectContinueTimeout: 1 * time.Second,
-	},
-	Timeout: defaultHTTPTimeout,
-}
-
```
Comment (darccio) [inline internal/datastreams/transport.go:52]: Is this dead code?

**Review by darccio — CHANGES_REQUESTED** [review]

File: ddtrace/tracer/metrics_test.go (hunk 3 above)
Comment (dianashevchenko) [reply ddtrace/tracer/metrics_test.go:16]: I ran the goimports, it's not me

File: ddtrace/tracer/option.go (hunk 4 above)
Comment (dianashevchenko) [reply ddtrace/tracer/option.go:28]: Fair

File: internal/datastreams/transport.go (hunk 6 above)
Comment (dianashevchenko) [reply internal/datastreams/transport.go:52]: Yes, wasn't used anywhere

File: ddtrace/tracer/metrics_test.go (hunk 3 above)
Comment (darccio) [reply ddtrace/tracer/metrics_test.go:16]: Oh, ok!

**Review by darccio — APPROVED** [review]

**Review by darccio — APPROVED** [review]
Approving again.

File: ddtrace/tracer/option.go [hunk 7]
```diff
@@ -851,6 +849,13 @@ func WithAgentAddr(addr string) StartOption {
 	}
 }
 
+// WithAgentTimeout sets the timeout for the agent connection. Timeout is in seconds.
+func WithAgentTimeout(timeout int) StartOption {
+	return func(c *config) {
+		c.httpClientTimeout = time.Duration(timeout) * time.Second
+	}
+}
```
Comment (darccio) [inline ddtrace/tracer/option.go:857]: Could this option function be unexported?

Comment (dianashevchenko) [reply ddtrace/tracer/option.go:857]: I think it's better for different products, including inside DD, to have it available. I was wary of adding an env variable, so I opted for the tracer option.

**Review by piochelepiotr — APPROVED** [review]

**Pull request merged**